ci/builders/README.md

```diff
@@ -154,231 +143,237 @@ An archive component is used to tell the recipes which artifacts are generated b
 
-The properties description is as follows:
+The property's description is as follows:
 
-* **Name**, the name of the step running the script.
-* **Parameters**, flags passed to the script. Both input and output paths must be relative to the checkout directory.
+* **Name** - the name of the step running the script.
+* **Parameters** - flags passed to the script. Both input and output paths must
+be relative to the checkout directory.
 * **Script**, the script path relative to the checkout repository.
 * **Language**, the script language executable to run the script. If empty it is assumed to be bash.
 
-    ```
-      {
-                "name": "Debug-FlutterMacOS.framework",
-                "parameters": [
-                    "--dst",
-                    "out/debug",
-                    "--arm64-out-dir",
-                    "out/ios_debug",
-                    "--simulator-x64-out-dir",
-                    "out/ios_debug_sim",
-                    "--simulator-arm64-out-dir",
-                    "out/ios_debug_sim_arm64"
-                ],
-                "script": "flutter/sky/tools/create_full_ios_framework.py",
-                "language": "python3"
-            }
-
+```yaml
+{
+    "name": "Debug-FlutterMacOS.framework",
+    "parameters": [
+        "--dst",
+        "out/debug",
+        "--arm64-out-dir",
+        "out/ios_debug",
+        "--simulator-x64-out-dir",
+        "out/ios_debug_sim",
+        "--simulator-arm64-out-dir",
+        "out/ios_debug_sim_arm64"
+    ],
+    "script": "flutter/sky/tools/create_full_ios_framework.py",
+    "language": "python3"
+}
 ```
 
+### Global Tests
 
+Global tests follow the same format as build tests but defined at the top
+level. The main difference from local tests is that global tests have access
+to the sub-build artifacts.
 
+### Global Generators
 
-#### Archives 
-
-The archives component provides instructions to upload the artifacts generated by the global generators. Is a list of dictionaries with two keys: source and destination. Source is a path relative to the checkout repository and destination is a relative path to &lt;bucket>/flutter/&lt;commit>.
-
+Global generators follow the same format as local generators but defined at
+the build top level. The main difference is that global generators can create
+new artifacts combining outputs of multiple sub-builds.
 
-```
+### Global Archives
 
-    "archives": [
-        {
-            "source": "out/debug/artifacts.zip",
-            "destination": "ios/artifacts.zip"
-        },
-        {
-            "source": "out/debug/ios-objcdoc.zip",
-            "destination": "ios-objcdoc.zip"
-        }
-    ]
+The archives component provides instructions to upload the artifacts generated
+by the global generators. Is a list of dictionaries with two keys: source and
+destination. Source is a path relative to the checkout repository and
+destination is a relative path to &lt;bucket>/flutter/&lt;commit>.
 
+```yaml
+"archives": [
```

Comment:
I'm a bit confused about how/why this differs from the `archives` list in sub-builds. The `archives` components in sub-builds had a lot more options. Do they need to also be referenced here at the top-level, or do they get captured automatically somehow? A more detailed description both here and there would be helpful.

Reply:
The archives section on the subbuilds is more complex because it supports scenarios like subbuilds uploading multiple jar and pom files to a bucket different from the one used for all the other archives and GN+ninja artifacts being generated directly inside the zip_archives folder of the build output. 

The global generator use case is simpler because the script have full control about where inside src/out to generate the artifacts and also have the freedom to specify the path relative to the bucket where the artifact will be uploaded to.

I have a pending action item to move the archives inside the global generator configuration to make them simpler to understand and potentially remove complexity required during the migration.